Piro Thyra Sensitivities

mperego opened the pull request:
Piro: Implemet adjoint sensitivities for Piro::NoxSolver (Thyra stack). Issue #3088

The adjoint system is solved by implicitly transposing the Jacobian operator and its preconditioner operator (thanks to @bartlettroscoe for the helping me with this).
This contains also a fix for update logic in ROL Thyra adapter, as well as a fix to allow Belos Thyra adapter to correctly select left/right preconditioner by @bartlettroscoe .

bartlettroscoe commented:
@mperego, is this ready to review and merge?

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

mperego commented:
@bartlettroscoe Yes it is ready for review.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1044</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 749</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 303</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1048</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 753</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 307</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (mperego/Trilinos)</dt>
  <ul>
    <li>Branch: PiroThyraSensitivities</li>
    <li>SHA: b099a574d7604d02eb19b86eb6ebc7be78369385</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: mperego
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1048</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 753</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 307</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1050</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 755</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 309</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (mperego/Trilinos)</dt>
  <ul>
    <li>Branch: PiroThyraSensitivities</li>
    <li>SHA: b099a574d7604d02eb19b86eb6ebc7be78369385</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: mperego
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1050</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 755</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 309</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>b099a574d7604d02eb19b86eb6ebc7be78369385</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>964412f1414994017cf270619b5fc493b21d8f06</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

rppawlo commented on packages/piro/test/Main_AnalysisDriver_Tpetra.cpp:
Are the values of the optimum checked somewhere in this test? Why just printed to screen here? Could we change this driver to fail the test if the values aren't correct?

rppawlo commented on packages/piro/test/MockModelEval_A_Tpetra.hpp:
Why the Comment about Charon? Should probably be removed.

rppawlo commented on packages/piro/src/Piro_NOXSolver.hpp:
Can you add doxygen comments to explain the options for DerivativeLayout?

mperego commented on packages/piro/test/MockModelEval_A_Tpetra.hpp:
Thanks, it's a copy and paste leftover. I'll get rid of it.

mperego commented on packages/piro/test/Main_AnalysisDriver_Tpetra.cpp:
This makes sense. I ported this driver from the Epetra version one, but it makes sense to have an error check. I'll put the error check in the Epetra driver as well.

mperego commented on packages/piro/src/Piro_NOXSolver.hpp:
Sure. thanks.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1062</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 767</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 321</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (mperego/Trilinos)</dt>
  <ul>
    <li>Branch: PiroThyraSensitivities</li>
    <li>SHA: 0584d83a41f5954be002be84f5b86af58143bdd2</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: mperego
</details>

mperego commented:
@rppawlo, I have rebased my branch to make it  cleaner but unfortunately this got rid of all your comments. Sorry about that. Anyway, I implemented your suggestiond. Also I fixed the way I was creating the Tpetra map in the Tpetra driver. Thanks for your comments!

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1062</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 767</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 321</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0584d83a41f5954be002be84f5b86af58143bdd2</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Error: Jenkins Jobs - A user has commited a change to the PR before testing completed.  The original testing SHA = 17086a23321a265e900f35fb91dc649cbf4a80e3 Does not match the current commit SHA = 9a6b8f13897f05aa1f3943e5a922492ec28347ee.  The Jenkins Jobs will be shutdown; Testing of this PR must occur again.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1068</li>
  <li>Status: ERROR</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>17086a23321a265e900f35fb91dc649cbf4a80e3</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 773</li>
  <li>Status: ERROR</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>17086a23321a265e900f35fb91dc649cbf4a80e3</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 327</li>
  <li>Status: ERROR</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>17086a23321a265e900f35fb91dc649cbf4a80e3</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1070</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 775</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 329</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (mperego/Trilinos)</dt>
  <ul>
    <li>Branch: PiroThyraSensitivities</li>
    <li>SHA: 9a6b8f13897f05aa1f3943e5a922492ec28347ee</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: mperego
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1070</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 775</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 329</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3055</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>PiroThyraSensitivities</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/mperego/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>9a6b8f13897f05aa1f3943e5a922492ec28347ee</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>026bc7cda9da67864f06e65ce2e8e5aa5760ebaf</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3055.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3055-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

mperego commented:
@rppawlo when you get a chance, can you re-approve this? Thanks, and sorry for the rebase.

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ rppawlo ]!

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request MUST BE MERGED MANUALLY BY Project Team - Master Automerge is disabled (in .cfg file)

bartlettroscoe commented:
@trilinos/framework,

FYI: This PR branch was not merged.  Instead, it appears that the 5 commits were rebased and pushed.  My push logger recorded the push:

```
commit a2e4adbd24235a996571f808bd2f53f5440b6bb5
Author:     Mauro Perego <mperego@sandia.gov>
AuthorDate: Tue Jul 10 13:33:01 2018 -0600
Commit:     mperego <mperego@sandia.gov>
CommitDate: Fri Jul 13 08:41:50 2018 -0600

    Piro: Add tests to exercise the newly implemented adjoint sensitivities capability (issue #3088)

Commits pushed:
a2e4adb Piro: Add tests to exercise the newly implemented adjoint sensitivities capability (issue #3088)
704fa6f Piro: Add adjoint sensitivities capability for Thyra Piro_NOXSolver (issue #3088)       The adjoint system is solved by implicitly transposing the jacobian operator       and the preconditioner operator (thanks to @bartlettroscoe).
1f4ef6d Piro: allow for ROL trust region method in Piro::PerformAnalysis.
e407a34 ROL: Thyra Adapter, fix update logic (related to issue #3088)
00b55c4 Belos: Allow for explicit left and right preconditioners (related to issue #3087)
```

These commits are on the first-parent history of 'develop'.  Therefore, git bisection that hits these intermediate commits will not be robust because they may likely fail.

If we would just turn off the configuration option "Allow rebase merging" (see #2726) then we could avoid this happening in the future.

For example, try to click the "Revet" button above.  GitHub tells you you can't do it.  

On the other hand, a PR branch that is merged with merge commit like #3100 can be reverted (just click the button).
